### PR TITLE
include vin in published message body

### PR DESCRIPTION
**Pull request by patrickdemers6**

### CODE REVIEW

**Comment by patrickdemers6** [comment]
Open to better ways to add the vin to the body. I couldn't figure out a cleaner way that worked with Go's type system.

File: telemetry/record.go [hunk 1]
```diff
@@ -95,3 +106,27 @@ func (record *Record) ensureEncoded() {
 		record.Serializer.Logger().Error("record_RawBytes_blank")
 	}
 }
+
+func getPayloadWithVin(record *Record) ([]byte, error) {
+	if record.TxType == "alerts" {
```
Comment (aaronpkahn) [inline telemetry/record.go:111]: minor: can we use a `switch` to improve readability and ease of adding new records?

File: telemetry/record_test.go [hunk 2]
```diff
@@ -31,4 +38,48 @@ var _ = Describe("Socket handler test", func() {
...
+		message := messages.StreamMessage{TXID: []byte("1234"), SenderID: []byte("vehicle_device.42"), MessageTopic: []byte("V"), Payload: generatePayload("cybertruck", "42", nil)}
+		recordMsg, err := message.ToBytes()
+		Expect(err).To(BeNil())
+
+		record, err := telemetry.NewRecord(serializer, recordMsg, "1")
+		Expect(record).To(Not(BeNil()))
+		Expect(err).To(BeNil())
+		Expect(record.Serializer).To(Not(BeNil()))
+
+		data := &protos.VehicleMetrics{}
```
Comment (aaronpkahn) [inline telemetry/record_test.go:61]: why do we use a `protos.VehicleMetrics` here if we use a `protos.Payload` in generatePayload?

File: telemetry/record_test.go [hunk 3]
```diff
@@ -5,9 +5,16 @@ import (
...
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"github.com/sirupsen/logrus/hooks/test"
+	// "google.golang.org/protobuf/proto"
 
+	"github.com/teslamotors/fleet-telemetry/messages"
+	"github.com/teslamotors/fleet-telemetry/protos"
+
+	// "github.com/teslamotors/fleet-telemetry/protos"
```
Comment (aaronpkahn) [inline telemetry/record_test.go:17]: remove comments

**Review by aaronpkahn — APPROVED** [review]

File: telemetry/record.go [hunk 4]
```diff
@@ -95,3 +106,28 @@ func (record *Record) ensureEncoded() {
 		record.Serializer.Logger().Error("record_RawBytes_blank")
 	}
 }
+
+func getPayloadWithVin(record *Record) ([]byte, error) {
```
Comment (agbpatro) [inline telemetry/record.go:110]: nit: how about something like:-
```sh
func (record *Record) getPayloadWithVin() ([]byte, error) {
	switch record.TxType {
	case "alerts":
		message := &protos.VehicleAlerts{}
		err := proto.Unmarshal(record.Payload(), message)
		message.Vin = record.Vin
		payload, _ := proto.Marshal(message)
		return payload, err
	case "errors":
		message := &protos.VehicleErrors{}
		err := proto.Unmarshal(record.Payload(), message)
		message.Vin = record.Vin
		payload, _ := proto.Marshal(message)
		return payload, err
	case "V":
		message := &protos.Payload{}
		err := proto.Unmarshal(record.Payload(), message)
		message.Vin = record.Vin
		payload, _ := proto.Marshal(message)
		return payload, err
	}

	return record.PayloadBytes, nil
}
```

Comment (patrickdemers6) [reply telemetry/record.go:110]: Not seeing the difference, I updated to use switch case earlier this morning. Is there something else I'm missing? @agbpatro

**Review by agbpatro — APPROVED** [review]

File: telemetry/record.go [hunk 5]
```diff
@@ -95,3 +104,28 @@ func (record *Record) ensureEncoded() {
...
 	}
 }
+
+func (record *Record) injectPayloadWithVin() error {
+	switch record.TxType {
+	case "alerts":
+		message := &protos.VehicleAlerts{}
+		err := proto.Unmarshal(record.Payload(), message)
+		message.Vin = record.Vin
+		record.PayloadBytes, _ = proto.Marshal(message)
```
Comment (agbpatro) [inline telemetry/record.go:114]: should we check for error instead of ignoring it?

File: telemetry/record.go [hunk 6]
```diff
@@ -36,7 +39,13 @@ func NewRecord(ts *BinarySerializer, msg []byte, socketID string) (*Record, erro
 	if err != nil {
 		return rec, err
 	}
-	return rec, nil
+
+	err = rec.injectPayloadWithVin()
+	if err != nil {
+		return rec, err
+	}
```
Comment (agbpatro) [inline telemetry/record.go:46]: not needed

File: telemetry/record.go [hunk 7]
```diff
@@ -36,7 +39,9 @@ func NewRecord(ts *BinarySerializer, msg []byte, socketID string) (*Record, erro
 	if err != nil {
 		return rec, err
 	}
-	return rec, nil
+
```
Comment (agbpatro) [inline telemetry/record.go:42]: nit:extra line?

**Review by agbpatro — APPROVED** [review]

**Pull request merged**